util/http/log/redact_http_query_values.go

```diff
@@ -0,0 +1,53 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package log
+
+import (
+	"errors"
+	"net/url"
+)
+
+// RedactHTTPQueryValuesFromError is a log utility to parse an error as a URL error and redact
+// HTTP query values to prevent leaking sensitive information like encoded credentials or tokens.
+func RedactHTTPQueryValuesFromError(err error) error {
+	var urlErr *url.Error
+
+	if err != nil && errors.As(err, &urlErr) {
+		url, urlParseErr := url.Parse(urlErr.URL)
+		if urlParseErr != nil {
```

Comment:
nit: can we leave the url as is and invert the condition

```go

url, urlParseErr := url.Parse(urlErr.URL)
if urlParseErr == nil { 
  RedactHTTPQueryValuesFromURL(url)
  urlErr.URL = url.Redacted()
  return urlErr
}
return err
```

Reply:
It's difficult. It seems to be a trade-off between security and maintainability. Could a malformed URL contain sensitive information? Maybe. Is it likely for SOCI to have malformed URLs? Probably not.

I guess it is fine to take the suggestion as it is simpler and continue to monitor.